File: gauss_kernel.py
import numpy as np

def gauss_kernel(sigma, sample_rate, duration):
    """
    Creates a Gaussian kernel centered in a vector of given duration.

    :param sigma: the standard deviation of the kernel in seconds
    :param sample_rate: the temporal resolution of the kernel in Hz
    :param duration: the desired duration of the kernel in seconds, (in general at least 4 * sigma)

    :return: the kernel as numpy array
    """
    l = duration * sample_rate
    x = np.arange(-np.floor(l / 2), np.floor(l / 2)) / sample_rate
    y = (1 / (np.sqrt(2 * np.pi) * sigma)) * np.exp(-(x ** 2 / (2 * sigma ** 2)))
    y /= np.sum(y)
    return y

File: gaussKernel.m
function y = gaussKernel(sigma, dt)
% y = gaussKernel( sigma, dt) creates a vector that 
% contains a Gaussian kernel in its center. The vector has the duration
% of 8 times the standard deviation.
%
% 'sigma' defines the standard deviation of the gaussian.
% 'dt' the temporal resolution, i.e. the stepsize or sampling interval
%
% by Jan Grewe, no warrenty!


x = -4 * sigma:dt:4 * sigma;
y = exp(-0.5 * (x ./ sigma) .^ 2) / sqrt(2 .* pi) ./ sigma;